cime/src/drivers/mct/main/prep_glc_mod.F90

```diff
@@ -1108,4 +1443,176 @@ function prep_glc_get_mapper_Fl2g()
+      pressureOffset = max(interfacePressure(iCell) - referencePressure, 0.0_r8)
+      T0 = SHR_CONST_TF0 + SHR_CONST_DTF_DP * pressureOffset
+         !Note: These two terms for T0 are not needed because we are evaluating at salinity=0:
+         !+ SHR_CONST_DTF_DS * oceanSalinity(iCell) + SHR_CONST_DTF_DPDS * pressureOffset * oceanSalinity(iCell)
+      iceDeltaT = T0 - iceTemperature(iCell)
+      dTf_dS = SHR_CONST_DTF_DS + SHR_CONST_DTF_DPDS * pressureOffset
+
+      transferVelocityRatio = oceanSaltTransferVelocity(iCell)/oceanHeatTransferVelocity(iCell)
+
+      a = -1.0_r8 * dTf_dS * (1.0_r8 + nu)
+      b = transferVelocityRatio*Tlatent - nu*iceDeltaT + oceanTemperature(iCell) - T0
+      c = -transferVelocityRatio*Tlatent*max(oceanSalinity(iCell), 0.0_r8)
+      ! a is non-negative; c is strictly non-positive so we never get imaginary roots.
+      ! Since a can be zero, we need a solution of the quadratic equation for 1/Si instead of Si.
+      ! Following: https://people.csail.mit.edu/bkph/articles/Quadratics.pdf
+      ! Since a and -c are are non-negative, the term in the square root is also always >= |b|.
+      ! In all reasonable cases, b will be strictly positive, since transferVelocityRatio*Tlatent ~ 2 C,
+      ! T0 ~ -1.8 C and oceanTemperature should never be able to get below about -3 C
+      ! As long as either b or both a and c are greater than zero, the strictly non-negative root is
+      outInterfaceSalinity(iCell) = max(-(2.0_r8*c)/(b + sqrt(b**2 - 4.0_r8*a*c)), minInterfaceSalinity)
+
+      outInterfaceTemperature(iCell) = dTf_dS*outInterfaceSalinity(iCell)+T0
+
+      outFreshwaterFlux(iCell) = SHR_CONST_RHOSW*oceanSaltTransferVelocity(iCell) &
+        * (oceanSalinity(iCell)/outInterfaceSalinity(iCell) - 1.0_r8)
+
+      ! According to Jenkins et al. (2001), the temperature fluxes into the ocean are:
+      !   1. the advection of meltwater into the top layer (or removal for freezing)
+      !   2. the turbulent transfer of heat across the boundary layer, based on the termal driving
+      outOceanHeatFlux(iCell) = SHR_CONST_CPSW*(outFreshwaterFlux(iCell)*outInterfaceTemperature(iCell) &
+        - SHR_CONST_RHOSW*oceanHeatTransferVelocity(iCell)*(oceanTemperature(iCell)-outInterfaceTemperature(iCell)))
+
+      ! the temperature fluxes into the ice are:
+      !   1. the advection of ice at the interface temperature out of the domain due to melting
+      !      (or in due to freezing)
+      !   2. the diffusion (if any) of heat into the ice, based on temperature difference between
+      !      the reference point in the ice (either the surface or the middle of the bottom layer)
+      !      and the interface
+      outIceHeatFlux(iCell) = -SHR_CONST_CPICE*outFreshwaterFlux(iCell)*outInterfaceTemperature(iCell)
+
+      outIceHeatFlux(iCell) = outIceHeatFlux(iCell) &
+        - iceHeatFluxCoeff*(iceTemperature(iCell) - outInterfaceTemperature(iCell))
+
+      if (iCell==72) then
```

Comment:
Please remove this block of diagnostic output.

Reply:
Fixed in 733a46925b5580e5888bdfa417add9916e7cf2df